src/lxc/start.c

```diff
@@ -222,6 +225,20 @@ int lxc_check_inherited(struct lxc_conf *conf, bool closeall,
 		    (i < len_fds && fd == fds_to_ignore[i]))
 			continue;
 
+		/* Keep state clients that wait on reboots. */
+		lxc_list_for_each(cur, &conf->state_clients) {
+			struct lxc_state_client *client = cur->elem;
```

Comment:
Seems like this should be a separate helper?

Reply:
I'm not sure this licenses a separate helper.